From 5a9ec4918bbd57553afe8efe7391a08fa97d758e Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Tue, 18 Sep 2018 19:31:24 +0800
Subject: [PATCH 4678/5242] MLK-19614 VPU: Set fixed video node number for vpu
 decoder and encoder

commit  dd136139ad2789d4fd26fc16771e8d8b3b958c38 from
https://source.codeaurora.org/external/imx/linux-imx.git

Set /dev/video12 for decoder and /dev/video13 for encoder

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c         |    2 +-
 drivers/mxc/vpu-decoder-b0/vpu_b0.h         |    1 +
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c |    2 +-
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.h |    1 +
 4 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index d566fd1..13dd881 100755
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -3009,7 +3009,7 @@ static int vpu_probe(struct platform_device *pdev)
 
 		if (video_register_device(dev->pvpu_decoder_dev,
 					VFL_TYPE_GRABBER,
-					-1)) {
+					DECODER_NODE_NUMBER)) {
 			vpu_dbg(LVL_ERR, "error: %s unable to register video decoder device\n",
 					__func__
 					);
diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.h b/drivers/mxc/vpu-decoder-b0/vpu_b0.h
index dbeda0b..a767241 100755
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.h
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.h
@@ -67,6 +67,7 @@
 #define V4L2_MAX_CTRLS 12
 #define V4L2_PIX_FMT_NV12_10BIT    v4l2_fourcc('N', 'T', '1', '2') /*  Y/CbCr 4:2:0 for 10bit  */
 #define INVALID_FRAME_DEPTH -1
+#define DECODER_NODE_NUMBER 12 // use /dev/video12 as vpu decoder
 
 struct vpu_v4l2_control {
 	uint32_t id;
diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
index 0e4bad6..5f15a48 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
@@ -2352,7 +2352,7 @@ static int vpu_probe(struct platform_device *pdev)
 
 		if (video_register_device(dev->pvpu_encoder_dev,
 					VFL_TYPE_GRABBER,
-					-1)) {
+					ENCODER_NODE_NUMBER)) {
 			vpu_dbg(LVL_ERR, "%s unable to register video encoder device\n",
 					__func__
 					);
diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.h b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.h
index 380cb79..128e2a0 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.h
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.h
@@ -63,6 +63,7 @@
 #define ENC_REG_BASE 0x2c000000
 #define MIN_BUFFER_COUNT 3
 #define V4L2_MAX_CTRLS 12
+#define ENCODER_NODE_NUMBER 13 //use /dev/video13 as encoder node
 struct vpu_v4l2_control {
 	uint32_t id;
 	enum v4l2_ctrl_type type;
-- 
1.7.9.5

